pkg/scheduler/scheduler.go

```diff
@@ -319,6 +319,35 @@ func (s *Scheduler) nominate(ctx context.Context, workloads []workload.Info, sna
 	return entries
 }
 
+// resourcesToReserve calculates how much of the available resources in cq/cohort assignment should be reserved.
+// For Peempt mode it makes sure to not reserve resources more than nominal quota.
+// TODO: we can make reservation more accurate, by accounting how many resources
+// the assignment is going to preempt.
```

Comment:
I don't think this will ever be necessary. We need to block the quota regardless of whether it will come from preemption or it fits within the unused quota.